internal/resolution/datasource/maven_registry.go

```diff
@@ -64,20 +78,25 @@ func (m *MavenRegistryAPIClient) WithoutRegistries() *MavenRegistryAPIClient {
 }
 
 // AddRegistry adds the given registry to the list of registries if it has not been added.
-func (m *MavenRegistryAPIClient) AddRegistry(registry string) error {
-	if slices.Contains(m.registries, registry) {
-		return nil
+func (m *MavenRegistryAPIClient) AddRegistry(registry MavenRegistry) error {
+	for _, reg := range m.registries {
+		if reg.URL == registry.URL {
+			return nil
+		}
```

Comment:
There's an edge case if you have two registries with the same URL but different enabled snapshots/releases.
It might be not too bad to double-up on URLs anyway since the duplicate requests should be cached.

Reply:
With using ID to check if a registry should be added, this edge case should be handled now since registries with the same URL may still be both added if the IDs are different.